cli/cloudproviders/openstack.py

```diff
@@ -0,0 +1,426 @@
+            msg = "Insufficient config related to Cloud provider"
+            log.error(msg)
+            raise ConfigError()
+
+        except Exception as e:
+            log.error(e)
+            raise CloudProviderError(e)
+
+    def get_node_by_name(self, name):
+        """Get node object by name
+
+        Args:
+            name (str): Name of node
+        """
+        try:
+            nodes = [node for node in self._driver.list_nodes() if node.name == name]
+        except Exception as e:
+            log.error(e)
+            raise CloudProviderError(e)
+
+        return nodes[0] if nodes else None
+
+    def get_node_id(self, node):
+        """Get node id
+
+        Args:
+            node (OpenstackNode): Openstack node object
+        """
+        try:
+            return node.id
+        except Exception as e:
+            log.error(e)
+            raise CloudProviderError(e)
+
+    def get_node_state_by_id(self, id):
+        """Get node status by id
+
+        Args:
+            id (str): id of node
+        """
+        try:
+            node = self._driver.ex_get_node_details(id)
+            return node.state if node else None
+        except Exception as e:
+            log.error(e)
+            raise CloudProviderError(e)
+
+    def get_nodes_by_prefix(self, prefix):
+        """Get list of nodes by prefix
+
+        Args:
+            prefix (str): Node name prefix
+        """
+        try:
+            return [
+                (n.name, n.id) for n in self._driver.list_nodes() if prefix in n.name
+            ]
+        except Exception as e:
+            log.error(e)
+            raise CloudProviderError(e)
+
+    def get_node_volumes(self, node):
+        """Get list of storage volume ids attached to node
+
+        Args:
+            node (OpenstackNode): Openstack node object
+        """
+        try:
+            return [v.get("id") for v in node.extra.get("volumes_attached")]
+        except Exception as e:
+            log.error(e)
+            raise CloudProviderError(e)
+
+    def wait_for_node_state(self, node, state, timeout, interval):
+        """Wait for node to be in state
+
+        Args:
+            node (OpenstackNode): Openstack node object
+            state (str): Expected node state
+            timeout (int): Operation waiting time in sec
+            interval (int): Operation retry time in sec
+        """
+        id = self.get_node_id(node)
+        log.info(f"Waiting for node '{id}' to be in '{state}' state")
+        for w in WaitUntil(timeout=timeout, interval=interval):
+            try:
+                _state = self.get_node_state_by_id(id)
+            except Exception as e:
+                log.error(e)
+                raise CloudProviderError(e)
+
+            if _state == state:
+                log.info(f"Node '{id}' is in exptected '{_state}' state")
+                break
+
+            msg = f"Node '{id}' is in expected '{_state}' state, retry after {interval} sec"
```

Comment:
I think this msg should be 
msg = f"Node '{id}' is not in expected '{_state}' state, retry after {interval} sec"